storage/include/Database.hpp

```diff
@@ -17,10 +17,17 @@ class Database {
     Database(const std::string& db_path);
     ~Database();
 
+    enum OnDuplicateInsertion {
+      IGNORE,
+      FAIL
+    };
+
     /// this is low-level logic (separate?)
     bool store(const std::string& hash, const std::string& pubKey,
                const std::string& bytes, uint64_t ttl, uint64_t timestamp,
-               const std::string& nonce);
+               const std::string& nonce, OnDuplicateInsertion behaviour = FAIL);
+
+    bool bulk_store(const std::vector<service_node::storage::Item>& items);
```

Comment:
I'm not sure if we want `Item` or `message_t` here. `Database::store` already computes `exp_time` so you shouldn't have to also pass it to `Database::bulk_store`. I guess could address that the same time as we merge `Item` / `message_t`.

Reply:
Ok, will leave it as is for now